Mathlib/RingTheory/Artinian.lean

```diff
@@ -498,4 +498,31 @@ instance isMaximal_of_isPrime (p : Ideal R) [p.IsPrime] : p.IsMaximal :=
       localization_surjective (nonZeroDivisors (R ⧸ p)) (FractionRing (R ⧸ p))⟩).isField _
     <| Field.toIsField <| FractionRing (R ⧸ p)
 
+open BigOperators in
+/--
+[Stacks Lemma 00J7](https://stacks.math.columbia.edu/tag/00J7)
+-/
+lemma maximal_ideals_finite : (setOf <| Ideal.IsMaximal (α := R)).Finite := by
```

Comment:
Several comments:

`setOf` is usually written `{I : Ideal R // I.IsMaximal}`.

In an artinian ring, all primes are maximal, so you can have a `I.IsPrime` version as well.

In an artinian ring (which is also noetherian), all primes are also minimal, so you can use #9088 to prove this.

Reply:
My plan of proving minimal primes = maximal primes in artinian ring is to prove artinian rings are 0-dimensional noetherian ring

Reply:
Okay, it seems the proof of artinian -> noetherian is complicated and uses the result you're proving here, so using #9088 to prove this is probably circular. But the [proof that all primes are maximal](https://math.stackexchange.com/a/27147/12932) (and hence all are minimal) is at least easy.

Reply:
In the dimension theory branch, it is already proved that artinian rings are 0-dimensional and prime ideals of zero dimensional ring are maximal.

Reply:
Actually the result is already in mathlia as [IsArtinianRing.isMaximal_of_isPrime](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Artinian.html#IsArtinianRing.isMaximal_of_isPrime) via #6309 and #6676 ... We've forgotten about our own PRs!

Reply:
And it's literally the lemma above this one. 🤣

Reply:
Back to the proof. I think it would be slightly easier to use [IsArtinian.set_has_minimal](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Artinian.html#IsArtinian.set_has_minimal) and intersections ([Finset.inf](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Finset/Lattice.html#Finset.inf)) rather than products. You can consider all finite intersections of primes and there will be a minimal one. Any prime must contain one of the finitely many primes occuring in the minimal one, and hence equal to it by the previous lemma.

BTW I don't understand why [Ideal.IsPrime.inf_le'](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Ideal/Operations.html#Ideal.IsPrime.inf_le') and prod_le require `s` nonempty. When `s` is empty both sides are false so they should still hold. (Will submit a PR unless you beat me.)

Reply:
I have changed to your version and removed nonempty hypothesis from `Ideal.IsPrime.inf_le`. Thanks for the proof!